GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -333,12 +366,115 @@ public void Clear()
             Invalidate(invalidateChildren: true);
         }
 
+        public void EndSelectionAtLoad()
+        {
+            _loadedToBeSelectedRevisionsCount = 0;
+            ToBeSelectedObjectIds = Array.Empty<ObjectId>();
+            _toBeSelectedGraphIndexes = null;
+        }
+
+        public void EnsureRowVisible(int row)
+        {
+            int countVisible = DisplayedRowCount(includePartialRow: false);
+            int firstVisible = FirstDisplayedScrollingRowIndex;
+            if (row >= 0 && (row < firstVisible || firstVisible + countVisible <= row))
+            {
+                FirstDisplayedScrollingRowIndex = row;
+            }
+        }
+
+        public ObjectId? GetFirstNotSelectedObjectId()
+        {
+            if (_loadedToBeSelectedRevisionsCount > 0 || ToBeSelectedObjectIds.Count == 0)
+            {
+                // At least one of the revisions was selected
+                return null;
```

Comment:
This looks kind of contrary to the method name - i.e., the name states the result is the "first not selected revision id". `null` in this case would mean that **all** revisions are selected. But according to the comment, `null` indicates that there is an existing selection.... 🤯

Reply:
How about something like this?
```diff
 
-        public ObjectId? GetFirstNotSelectedObjectId()
-        {
-            if (_loadedToBeSelectedRevisionsCount > 0 || ToBeSelectedObjectIds.Count == 0)
-            {
-                // At least one of the revisions was selected
-                return null;
-            }
-
-            return ToBeSelectedObjectIds[0];
-        }
+        public bool HasExistingSelection => _loadedToBeSelectedRevisionsCount > 0 || ToBeSelectedObjectIds.Count == 0;
 
```

```diff
                     // All revisions are loaded (but maybe not yet the grid)
-                    if (_gridView.GetFirstNotSelectedObjectId() is ObjectId notSelectedId)
+                    if (!_gridView.HasExistingSelection)
                     {
+                        ObjectId notSelectedId = _gridView.ToBeSelectedObjectIds.First();
+
                         IEnumerable<ObjectId> parents = null;
```

Reply:
Changed this slightly.
Tho match the API, another check had to be added, to make sure the added check was not misused, GetGraphIndexes() needed a Assert too.